### DISCUSSION TITLE
Able to flash ESP8266 module, but on reboot, very fast flashing of LED, and endless random junk on console?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16285] [ESP8266]
Able to flash ESP8266 module, but on reboot, very fast flashing of LED, and endless random junk on console?  May be endless rebooting?  I flashed via Thonny, Tried 4MB image, and 1MB image same result.

**Comment by Jibun-no-Kage** [comment]
Seems to flash fine... 
C:\Users\jibun\AppData\Local\Temp\tmpcvq35aj9\ESP8266_GENERIC-FLASH_512K-20220618-v1.19.1.bin
esptool.py v4.7.0
Serial port COM4
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 84:0d:8e:8f:f5:b6
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Erasing flash (this may take a while)...
Chip erase completed successfully in 8.4s
Compressed 495084 bytes to 331639...
Writing at 0x00000000... (4 %)
Writing at 0x00005763... (9 %)
Writing at 0x0000c226... (14 %)
Writing at 0x000120b9... (19 %)
Writing at 0x00018c43... (23 %)
Writing at 0x0001ec10... (28 %)
Writing at 0x00024203... (33 %)
Writing at 0x00029cfb... (38 %)
Writing at 0x0002f9f0... (42 %)
Writing at 0x000349dc... (47 %)
Writing at 0x00039e10... (52 %)
Writing at 0x0003f809... (57 %)
Writing at 0x00045608... (61 %)
Writing at 0x0004b29c... (66 %)
Writing at 0x00050f0b... (71 %)
Writing at 0x0005717d... (76 %)
Writing at 0x0005d24d... (80 %)
Writing at 0x00063a3b... (85 %)
Writing at 0x000698a7... (90 %)
Writing at 0x00070ee5... (95 %)
Writing at 0x00077795... (100 %)
Wrote 495084 bytes (331639 compressed) at 0x00000000 in 29.9 seconds (effective 132.3 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
Done!

**Reply by robert-hh** [reply]
Try other flash modes. The default is DIO, but sometimes it has to be different, like DOUT. Flash modes can be set with Thonny.

**Comment by Jibun-no-Kage** [comment]
Yup... going to do that next...  Just wondered if something changed.  Stepped away from ESP8266/ESP32 projects for while, trying to catch back up.

**Comment by Jibun-no-Kage** [comment]
Well... I got some of my ESP2866 modules to updated MicroPython 1.24 via direct esptool on a Linux system.  Using the latest esptool verison.
```
# esptool.py version
esptool.py v4.8.1
4.8.1
```

However, I cannot seem to get any of my Wemos Mini D1 (ESP12F) based boards to flash.  They once flashed because they have version 1.19 on them, but now I just get no serial communication errors.

I even tried GPIO0 to GND with a jumper.

**Comment by Jibun-no-Kage** [comment]
Finally... So for any one that reads this later, for the Blue D1 Mini boards I have must do the following:

1) Must use TX and RX UART To USB FTDI Interface, For whatever reason, have to use UART Pins, i.e. TX and RX
2) Jumper D3 (GPIO0) To GND
3) Insert USB Power, NOT USB FTDI Communication, This must be done between each command
4) Flash at 460800 Baud, Slow Better In This Case

Command Sequence
1) Read MAC To Test Communication
```
# esptool.py --port /dev/ttyUSB0 --baud 460800 read_mac
```

2) Erase Memory
```
# esptool.py --port --baud 460800 /dev/ttyUSB0 erase_flash
```

3) Flash Firmware
```
# esptool.py --port /dev/ttyUSB0 --baud 460800 write_flash --flash_size=detect 0 ESP8266_GENERIC-20241025-v1.24.0.bin
```
Issues....
1) Could not, as in the past, get Windows 11, or Linux system to see device USB connection alone, had to use separate UART To FTDI interface, and USB only used for power.
2) Every insert of power, GPIO00 To GND Required Jumper, only one command per power up this way
3) Baud Rate Slow Down Required

Here is the mystery.... when I flashed MicroPython 1.19... some time ago on these same boards, I did not have any issues with 4 of the 5, and only 1 I had to use the GPIO00 To GND to force flash mode, and I did all 5 via Thonny from Windows 10.  But on Windows 11, no joy.  Not saying Windows 11 is the issue, I can flash various ESP8266 and ESP32 modules just fine via Thonny on Windows 11... just that these specific Blue D1 Mini (clones?) are picker now than in the past.

**Comment by robert-hh** [comment]
> these specific Blue D1 Mini (clones?)

Looks like you got bad clones. I never had any issues with Wemos ESP8266 boards,

**Comment by Jibun-no-Kage** [comment]
They work... just quirky to flash.  One of the 5, I have to pull the GPIO0 to GND after the flash sequence starts, then it communicates fine!  That is odd. The other 4, I can flash with GPIO0 to GND remaining in place.  GPIO0 to GND needed begs the 'design' question, they maybe just designed as such, since they have no explicit 'flash' button.  But the 1 that I have to pull the jumper for it to really go into flash mode, at the start of communication... that suggests something more to be sure.  I remember some of the first ESP8266 IoT devices I got years ago, need the same trick, you had to pull the GPIO0 to GND jumper at the initial communication for them to actually go into flash mode.
